src/test/java/com/klodnicki/Bike/v2/service/UserServiceHandlerTest.java

```diff
@@ -0,0 +1,134 @@
+package com.klodnicki.Bike.v2.service;
+
+import com.klodnicki.Bike.v2.DTO.user.UserForAdminResponseDTO;
+import com.klodnicki.Bike.v2.model.entity.User;
+import com.klodnicki.Bike.v2.repository.UserRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.modelmapper.ModelMapper;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class UserServiceHandlerTest {
+
+    private UserRepository userRepository; // to jest mock
+    private UserServiceHandler userServiceHandler;
+    private ModelMapper modelMapper;
+
+    @BeforeEach
+    public void setUp() {
+        userRepository = mock(UserRepository.class);
+        modelMapper = new ModelMapper();
+        userServiceHandler = new UserServiceHandler(userRepository, modelMapper);
+    }
+
+    @Test
+    public void add_ShouldReturnUserForAdminResponseDTO_WhenUserProvided() {
+        //Arrange
+        User user = new User();
+        when(userRepository.save(user)).thenReturn(new User());
+
+        UserForAdminResponseDTO expected = modelMapper.map(user, UserForAdminResponseDTO.class);
+
+        //Act
+        UserForAdminResponseDTO actual = userServiceHandler.add(user);
+
+        //Assert
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void findAll_ShouldReturnListOfUserForAdminResponseDTO_WhenProvidedIterableUsers() {
+        //Arrange
+        Iterable<User> iterableOfUsers = new ArrayList<>();
+        List<UserForAdminResponseDTO> expected = new ArrayList<>();
+
+        for (int i = 0; i < 3; i++) {
+            User user = new User();
+            ((ArrayList<User>) iterableOfUsers).add(user);
+        }
+
+        when(userRepository.findAll()).thenReturn(iterableOfUsers);
+
+        for (User user : iterableOfUsers) {
+            UserForAdminResponseDTO userDTO = modelMapper.map(user, UserForAdminResponseDTO.class);
+            expected.add(userDTO);
+        }
```

Comment:
This code is messy and hard to get grasp of. Could you please write it in a cleaner manner?

Reply:
I don't know what is messy here :) 
It's a reflection of original method findAll() ;)
![image](https://github.com/BartoliniAndBorderCollies/Bike.v2_REST_API/assets/126821059/e4e3e391-1da6-48e1-92ec-3bb847933c39)

First I create `List` and `Iterable`, then I create 3 users and I add them to `Iterable`.
Then I mock a method on repo so that it returns me Iterable of users. 
Next step is mapping each user to DTO.
Finally I call the original method and assert the outcome.

If it is difficult to read then I made some changes. I’ve extracted the logic for creating users and mapping users to DTOs into separate methods (createUsers and mapUsersToDTO). I think this makes the code more modular, easier to read, and easier to maintain. The reader can now focus on the logic of the test, rather than the details of how users and DTOs are created.

I hope that is fine!

Reply:
Okay, this looks fine now!